src/metadata.ts

```diff
@@ -96,10 +97,20 @@ export interface ApiObjectMetadata {
 
   /**
    * Additional metadata attributes.
-   * @jsii ignore
-   * @see https://github.com/cdk8s-team/cdk8s-core/issues/1297
    */
-  readonly [key: string]: any;
+  readonly additionalAttributes?: { [key: string]: any };
+}
+
+/**
+ * Options for `ApiObjectMetadataDefinition`.
+ */
+export interface ApiObjectMetadataDefinitionOptions extends ApiObjectMetadata {
+
+  /**
+   * Which ApiObject instance is the metadata attached to.
+   */
+  readonly apiObject: ApiObject;
```

Comment:
This is needed because it is what gives us access to the `App` instance during resolution, out of which we can extract the list of resolvers to use.